contents/performance/PerformanceProfiling.md

```diff
@@ -0,0 +1,176 @@
+# Performance Profiling
+
+Author: [Ong Heng Le](https://github.com/initialshl)
+
+<br>
+
+# Overview
+
+Software profiling is collecting data about a program's execution, by sampling statistical 
+data during the program's execution or by modifying the program itself. Profilers may 
+collect data such as function timings, memory allocation and CPU usage in approximated 
+or exact details. 
+
+Performance profiling mostly involves sampling the CPU and collecting function timings 
```

Comment:
I think the two paragraphs are wrong way round (put this paragraph as the intro) Was confused in the first paragraph because there's no mention of performance at all.